translator/model_enum.go

```diff
@@ -17,25 +17,8 @@ type CEnumSpec struct {
 }
 
 func (c *CEnumSpec) PromoteType(v Value) *CTypeSpec {
-	var (
-		int32Spec = CTypeSpec{Base: "int"}
-		int64Spec = CTypeSpec{Base: "long int"}
-	)
-	switch c.Type {
-	case int32Spec: // need promotion
-		switch v.(type) {
-		case int64:
-			c.Type = int64Spec
-		}
-	case int64Spec:
-	default:
-		switch v.(type) {
-		case int32:
-			c.Type = int32Spec
-		case int64:
-			c.Type = int64Spec
-		}
-	}
+	// ANSI C requires int values for enum. Therefore, force enum type to be int.
```

Comment:
**NOTE:**
Originally I implemented enums mapping by reading ANSI C, etc.
But there was some discrepancy between Go enums and C enums mapping, not sure if it came from CC or CGO.

I surely can't remember the details, e.g. there was a related bugfix in CCv1:
https://gitlab.com/cznic/cc/-/commit/a666f2ddc99ed82af6ce85864324c283c8b2f554

Reply:
Thanks for the feedback! And thanks for merging right away! Some fixes may still be required, but I'm sure we'll get this done.

Regarding this specific code: I was not able to replicate the existing behavior in cc v4, because Value type is always (u)int64 for numeric values with cc v4 (see e.g. https://pkg.go.dev/modernc.org/cc/v4#AdditiveExpression.Value).

Changes to the Value type are also kind of a problem in the macro translation code in translator.go. I did not test all of this (my examples only had few macros), so there might still be issues where Value was nil in v1. nil is no longer returned in v4, I tried to resolve this but there may be specific cases I did not consider.

Reply:
> because Value type is always (u)int64 for numeric values with cc v4

Yeah, this could be a problem. The type on the C side is dynamic and is depending on enums actual values, i.e. if the maximum value represented doesn't overflow `int8`, it will be used, so the Go counterpart (int64?) will not be compatible.

I really need to re-run this against my older bindings because they provided quite wide coverage of features. Maybe some changes to CCv4 will be required.

Reply:
> I really need to re-run this against my older bindings because they provided quite wide coverage of features. Maybe some changes to CCv4 will be required.

cznic adviced to use the Kind() method of the type object in v4 instead of the Value type (which was used in v1). See https://gitlab.com/cznic/cc/-/issues/137
However, I did not have any example on what is intended by the previous code, and since ANSI C requires enums to be ints, I did not consider any further.

Also, cznic strongly suggested to evaluate enums before translation due to different overflow behaviour in C and Go, see https://gitlab.com/cznic/cc/-/issues/136.

Let me know if I can help!

Reply:
Is there still anything open? Did you note specific cases which still cause problems? (More unit tests could be added...)
Otherwise, I'd be moving on to try and support "#pragma once"...